app/components/sendMailComponent.ts

```diff
@@ -0,0 +1,34 @@
+import {Component} from "angular2/core";
+import {MailService} from "./mailService"
+import {Router} from "angular2/router"
+import {EmailObject} from "./emailObject"
+import {Route} from "./routeConst"
+@Component({
+    templateUrl: "app/components/sendMailComponent.html"
+})
+export class SendMailComponent {
+    public email: EmailObject;
+    public mailService: MailService;
+    public router: Router;
+
+    constructor(mailService: MailService, router: Router) {
+        this.mailService = mailService;
+        this.router = router;
+        this.email = new EmailObject();
+        let self = this;
+
+    }
+    public sendMailClicked(emailItem: EmailObject) {
+        let self=this;
+        console.log(emailItem);
+        
+        this.mailService.sendMail(emailItem).then(function (mail: EmailObject) {
```

Comment:
you assigned self = this, so need to use self,mailService instance this.mailService